src/api/app/helpers/webui/staging/project_helper.rb

```diff
@@ -6,4 +6,22 @@ def icon_for_checks(checks, missing_checks)
     return 'fa-check-circle text-primary' if checks.all?(&:success?)
     'fa-exclamation-circle text-danger'
   end
+
+  def unstaged_staged_request_history(history)
+    capture do
+      concat(link_to("##{history.bs_request.number}", request_path(history.bs_request.number)))
+      concat(' for package ')
+      concat(content_tag(:b, history.package_name))
+      concat(' submitted by ')
+      concat(content_tag(:b, history.user_name))
+    end
+  end
+
+  def staging_project_created_history(history)
```

Comment:
Why a helper and not a partial?

Reply:
helpers are much faster?

Reply:
They are both really small, so I think using a helper is fine in this case. I don't have a strong opinion on this, I'm also fine with using a partial so I  can change it if you think it's a blocker.

Reply:
Lesson learned from refactoring repositories page: if the partial is small and repeated often, it should be a helper.

Reply:
"much faster" without benchmark is just a guess. I think the partial is easier to maintain, beside it's the "Rails way" https://apidock.com/rails/ActionView/Helpers/TextHelper/concat

Reply:
> I don't have a strong opinion on this

Me neither. If it's because of performance and it helps, so be it.

Reply:
I pushed changes to use a partial. Please have a look.